react/src/pages/utils/typings.tsx

```diff
@@ -151,6 +151,21 @@ export interface DataEntryHeader {
     hidden?: boolean;
 }
 
+export interface User {
+    username: string;
+    email: string;
+    isAdmin: boolean;
+    password?: string;
+    confirmPassword?: string;
+    groupMemberships?: string[];
+    deactivated?: boolean;
+}
+
+export interface UserAction {
+    type: "set" | "update" | "add" | "delete";
+    payload: User | User[];
+}
```

Comment:
As far as I can tell this is only used in UserList and specific to its internal state management. Are there plans that require this to be freestanding?